polaris-react/src/components/Frame/components/ContextualSaveBar/ContextualSaveBar.module.scss

```
@@ -70,12 +70,16 @@
   display: flex;
   flex-direction: row;
   gap: var(--p-space-200);
+  overflow: hidden;
+  margin-right: var(--p-space-200);
 
   // stylelint-disable -- Icon overrides specific to Contextual Save Bar
-  & [class*='Polaris-Icon__Svg'],
-  & [class*='Icon-Svg'] {
+  & [class*='Polaris-Icon__Svg'] {
     fill: var(--p-color-text-inverse);
   }
+  & [class*='Polaris-Icon'] {
+    flex-shrink: 0; /* Prevents the icon from shrinking */
```

```suggestion
    flex-shrink: 0;
```

This should be pretty understandable without the comment. Thanks @sainihas